powerauth-push-server/src/main/java/io/getlime/push/service/batch/UserDeviceItemReader.java

```diff
@@ -46,9 +46,9 @@ public UserDeviceItemReader(EntityManagerFactory entityManagerFactory, PushServi
         // Configure queries and reader
         this.setEntityManagerFactory(entityManagerFactory);
         this.setQueryString("select " +
-                " new io.getlime.push.repository.model.aggregate.UserDevice(c.userId, d.id, d.activationId, c.campaignId, c.appId, d.platform, d.pushToken) " +
+                " new io.getlime.push.repository.model.aggregate.UserDevice(d.userId, d.id, d.activationId, c.campaignId, d.appCredentials.id, d.platform, d.pushToken) " +
```

Comment:
It could be nice to extract this to a named query (not in this PR), if possible, to validate syntax at start of persistence context.

Reply:
For topics like this, open issue instead of commenting.